tools/integration_tests/tokenserver/test_e2e.py

```diff
@@ -215,14 +215,14 @@ def test_valid_oauth_request(self):
         raw = urlsafe_b64decode(res.json['id'])
         payload = raw[:-32]
         signature = raw[-32:]
-        payload_dict = json.loads(payload.decode('utf-8'))
+        payload_str = payload.decode('utf-8')
+        payload_dict = json.loads(payload_str)
         # The `id` payload should include a field indicating the origin of the
         # token
         self.assertEqual(payload_dict['tokenserver_origin'], 'rust')
         signing_secret = self.TOKEN_SIGNING_SECRET
-        expected_token = tokenlib.make_token(payload_dict,
-                                             secret=signing_secret)
-        expected_signature = urlsafe_b64decode(expected_token)[-32:]
+        tm = tokenlib.TokenManager(secret=signing_secret)
+        expected_signature = tm._get_signature(payload_str.encode('utf8'))
```

Comment:
This was fun to track down, the `tokenlib.make_token` serializes its input to json using [`json.dumps`](https://github.com/mozilla-services/tokenlib/blob/91ec9e2c922e55306eddba1394590a88f3b10602/tokenlib/__init__.py#L108).

Turns out `json.dumps` keeps some spaces in the serialized json, so if `serde_json` would have spit out `{"foo":"bar"}`, `json.dumps` dumps `{"foo": "bar"}` note the extra space!

So the extra spaces end up making it so that my implementation produces different signatures!